libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -498,8 +499,9 @@ bool _IotMqtt_DecrementOperationReferences( _mqttOperation_t * pOperation,
         EMPTY_ELSE_MARKER;
     }
 
-    /* Decrement job reference count. */
-    if( taskPoolStatus == IOT_TASKPOOL_SUCCESS )
+    /* Decrement job reference count only when either the connect operation has been completed in taskpool
+     *  or the job reference needs to be decreased after the waitable operation is completed i.e cancelJob parameter is false*/
+    if( ( taskPoolStatus == IOT_TASKPOOL_SUCCESS ) && ( ( pOperation->u.operation.type == IOT_MQTT_CONNECT ) || ( cancelJob == false ) ) )
```

Comment:
I am not sure, if I understand it right. Are you saying
1. Check if the operation is connect and if so check if taskpoolStatus is success OR
2. cancelJob is true.?

Is this above the correct statement of this check? If so, the condition doesn't seems correct.